File: altgraph_web_app/src/main/java/org/cjoakim/cosmos/altgraph/data/common/model/npm/Library.java
package org.cjoakim.cosmos.altgraph.data.common.model.npm;

import com.azure.spring.data.cosmos.core.mapping.Container;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.HashMap;

/**
 * Instances of this class represent a NPM (Node.js Package Manager) Library.
 * <p>
 * Chris Joakim, Microsoft, November 2022
 */

@Data
@Container(containerName = "npm_graph")
@JsonIgnoreProperties(ignoreUnknown = true)
@EqualsAndHashCode(callSuper = false)
public class Library extends NpmDocument {

    public Library() {
        super();
    }

    private String name;
    private String desc;
    private String[] keywords;
    private HashMap<String, String> dependencies;
    private HashMap<String, String> devDependencies;
    private String author;
    private String[] maintainers;
    private String version;
    private String[] versions;
    private String homepage;

    private int library_age_days;
    private int version_age_days;

    @Override
    public void intern() {

        baseIntern();
    }

}

File: altgraph_web_app/Models/Library.cs
namespace altgraph_web_app.Models
{
  public class Library : Entity
  {
    public string Name { get; set; } = string.Empty;
    public string Desc { get; set; } = string.Empty;
    public List<string> Keywords { get; set; } = new List<string>();
    public Dictionary<string, string> Dependencies { get; set; } = new Dictionary<string, string>();
    public Dictionary<string, string> DevDependencies { get; set; } = new Dictionary<string, string>();
    public string Author { get; set; } = string.Empty;
    public List<string> Maintainers { get; set; } = new List<string>();
    public string Version { get; set; } = string.Empty;
    public List<string> Versions { get; set; } = new List<string>();
    public string Homepage { get; set; } = string.Empty;
    public int LibraryAgeDays { get; set; }
    public int VersionAgeDays { get; set; }
  }
}